core/utils.go

```diff
@@ -301,3 +303,36 @@ func FileExists(path string) (bool, error) {
 
 	return false, err
 }
+
+// get body text with a text editor form env
+func getBodyTextWithEditor() (string, error) {
```

Comment:
```sh
EDITOR=nvim go run main.go new --dry-run --editor
✔ Kind … Added
✔ Issue … 513
kind: Added
time: 2023-08-05T01:03:30.8351218-07:00
custom:
  Issue: "513"
```

Hmm, it does not seem to be adding a body to the output here, am I missing something?

Reply:
i tried on a different laptop it's working.
I did the following 
```bash
git clone https://github.com/wizsk/changie
cd changie
git checkout allow-editor-to-create-body-msg
EDITOR=nvim go run main.go new --dry-run --editor
```

Reply:
What is the full output after you type in the body and issue?

Reply:
```bash
EDITOR=nvim go run main.go new --dry-run --editor
✔ Kind … Added
✔ Issue … 512
kind: Added
body: body msg
time: 2023-08-05T14:39:48.75477911+06:00
custom:
  Issue: "512"
```

Reply:
Weird, maybe I am just sleepy, it seems to me that the files are left in the temp folder and they are still empty after I save and exit. Not sure why though... anyway, will review again tomorrow, thanks for the hard work btw.

Reply:
I'm new to all these so you are really helping me a lot.

Reply:
with `EDITOR=gnome-text-editor go run main.go new -d -e` i got the same issue.
i changed io.ReadAll()` to `os.ReadFile()` now I think it's fixed.

Reply:
Hmm, still no difference for me, can't seem anything wrong in the code but it is a little confusing for sure.

Reply:
ok I was able to get some hints from the gh cli tool here https://github.com/cli/cli/blob/f777bec798601d715c4f0d05638c0f941c790d5b/pkg/surveyext/editor_manual.go#L43
using this package `shellquote "github.com/kballard/go-shellquote"`

Here is what I changed in the code to get a proper output:
On line 323 in utils.go
```go
args, err := shellquote.Split(editor)
if err != nil {
    return "", err
}
args = append(args, bodyTxtFile.Name())

// #nosec G204
cmd := exec.Command(args[0], args[1:]...)
```

My output looks like this:
```sh
$ EDITOR=nvim go run main.go new --dry-run --editor
✔ Kind … Added
✔ Issue … 512
kind: Added
body: "can add markdown here\r\n```go\r\nvar x = 5\r\nstrings.Repeat(x, 15)\r\n```"
time: 2023-08-05T15:49:19.5151001-07:00
custom:
  Issue: "512"
```

I think the core issue I was having was that this line of code:
```go
cmd := exec.Command("sh", "-c", editor+" "+bodyTxtFile.Name())
```
I don't think its necessary to use `sh -c` here and is actually causing problems, at least on windows.
However, simply doing `exec.Command(editor, bodyTxtFile.Name())` won't work if our editor has multiple arguments like with vscode. Which is where the shellquote package helps.

I think we should add the file extension of the version file to the temporary file so your editor can use any language configs you have configured, you can get that value by looking at the config.VersionExtension.
```go
change.Body, err = getBodyTextWithEditor(ctx.config.VersionExt)
```
This gives me a pretty nice experience with neovim and should carry to other editors.

Reply:
I should have thought about windows users. I'm really sorry for causing you this much hassle.

Reply:
Nah its all good, doing good work.

Reply:
Thanks for being kind. I will make the nessesary changes you asked for. 
I didn't quite understand the extension part.

Reply:
Oh yea for the extension, if you specify a file extension when creating the temp file, the passed in EDITOR will most likely have a plugin or extension for things like syntax highlighting or auto completion. For example, changie uses a markdown file for the release notes / changelog, eg "CHANGELOG.md". So if I pass in --editor I would want my temp file to be something like "random-file.md" so my editor will give me syntax highlighting for markdown.

The code for this looks like this:
```go
bodyTxtFile, err := os.CreateTemp("", "changie-body-txt-*."+ext)
```
( once you pass in the extension to the function it self I used `ext string` where ext comes from config.VersionExtension )

Reply:
ok Thanks.

Reply:
should I add this too?
```go
func needsBom() bool {
	// The reason why we do this is because notepad.exe on Windows determines the
	// encoding of an "empty" text file by the locale, for example, GBK in China,
	// while golang string only handles utf8 well. However, a text file with utf8
	// BOM header is not considered "empty" on Windows, and the encoding will then
	// be determined utf8 by notepad.exe, instead of GBK or other encodings.

	// This could be enhanced in the future by doing this only when a non-utf8
	// locale is in use, and possibly doing that for any OS, not just windows.

	return runtime.GOOS == "windows"
}
```

```go
	if needsBom() {
		if _, err := f.Write(bom); err != nil {
			return "", err
		}
	}
```

Reply:
btw my output when i Edit a file with new lines those looks like this
```bash
✔ Kind … Added
✔ Issue … 101
kind: Added
body: |-
  line no 1
  ``go
  fmt.Println("hi")
  ``
  line 5
time: 2023-08-06T05:59:36.589033325+06:00
custom:
  Issue: "101"
```

Reply:
Yea that is what I was hoping the output would be, maybe its the missing BOM on windows so yea I would probably add that.

Reply:
It could also be the different line endings where windows uses `\r\n` and not `\n`, not sure.

Reply:
Ok

Reply:
I added the things.
I have tested on windows with nodepad. Now everything veryting looks ok.

Reply:
we could use `strings.ReplaceAll(str, "\r\n", "\n")` to remove the extra `\r` ?

Reply:
I think yaml and/or git will normalize that either way, so I think its fine. Will see if anything happens later on.

Reply:
I think the only task left is to write the testsband the docs.
I haven't written much tests before. So what should I do?

Reply:
Hmm, for docs I think we are good with that CLI argument which will be autogenerated to the website. You will need a change fragment for your feature of course. For tests, maybe look around at the existing tests and I can review after you give it a shot. Or if you would rather do some live coding on discord or if you have another preference let me know.

Reply:
I can do live coding on discord.

Reply:
My name is miniscruff on discord if you wanted to reach out